homeassistant/components/tellduslive/config_flow.py

```diff
@@ -0,0 +1,133 @@
+            }))
+
+    async def async_step_auth(self, user_input=None):
+        """Handle the submitted configuration."""
+        if user_input is not None:
+            if self._session.authorize():
+                host = self._host or CLOUD_NAME
+                return self.async_create_entry(
+                    title=host,
+                    data={
+                        KEY_HOST: host,
+                        KEY_TOKEN: self._session.access_token
+                    } if self._host else {
+                        KEY_TOKEN: self._session.access_token,
+                        KEY_TOKEN_SECRET: self._session.access_token_secret
+                    },
+                )
+
+        async def get_auth_url(session):
+            return session.authorize_url
+
+        try:
+            with async_timeout.timeout(10):
+                auth_url = await get_auth_url(self._session)
+        except asyncio.TimeoutError:
+            return self.async_abort(reason='authorize_url_timeout')
+        except Exception:  # pylint: disable=broad-except
+            _LOGGER.exception("Unexpected error generating auth url")
+            return self.async_abort(reason='authorize_url_fail')
+
+        _LOGGER.debug('Got authorization URL %s', auth_url)
+        return self.async_show_form(
+            step_id='auth',
+            description_placeholders={
+                'app_name': APPLICATION_NAME,
+                'auth_url': auth_url,
+            },
+        )
+
+    async def async_step_discovery(self, user_input):
+        """Run when a Tellstick is discovered."""
+        from tellduslive import supports_local_api
+        _LOGGER.info('Discovered tellstick device: %s', user_input)
+        # Ignore any known devices
+        for entry in self._async_current_entries():
+            if entry.data[KEY_HOST] == user_input[0]:
+                return self.async_abort(reason='already_configured')
+
+        if not supports_local_api(user_input[1]):
+            _LOGGER.debug('Tellstick does not support local API')
+            # Configure the cloud service
+            return await self.async_step_auth()
+
+        self._hosts.append(user_input[0])
+        return await self.async_step_user()
+
+    async def async_step_import(self, user_input):
+        """Import a config entry."""
+        for entry in self._async_current_entries():
+            if (entry.title == user_input[KEY_HOST]
+                    or entry.title == CLOUD_NAME):
+                return self.async_abort(reason='already_configured')
+
+        if user_input[KEY_HOST] != DOMAIN:
+            self._hosts.append(user_input[KEY_HOST])
+
+        if not await self.hass.async_add_job(
+                os.path.isfile, self.hass.config.path(TELLDUS_CONFIG_FILE)):
+            return await self.async_step_user()
+
+        conf = await self.hass.async_add_job(
+            load_json, self.hass.config.path(TELLDUS_CONFIG_FILE))
+        host = next(iter(conf))
```

Comment:
Yes, we only store one entity in the json file (the idea was to support multiple entities but we kinda lost interest as neither me nor @molobrakos have more than one tellstick), it can either be `tellduslive` or a host.

With config_entities we no longer need a json so my idea was to remove the json file and after a few versions of hass make that option deprecated (so most users would be automatically upgraded) perhaps this is an unnecessary path.

Reply:
Ah, ok. That sounds good to me!